### ISSUE TITLE
Wrong array schema

### ISSUE THREAD

**Issue by bangbang93** [szokodiakos/typegoose#193]
When I build a schema like
```typescript
export interface IRoomSchema {
  _id: Types.ObjectId
  name: string
  type: EnumRoomType
  ownerId: number
  avatar: string
  users?: IRoomUserSchema[]
}

export interface IRoomUserSchema {
  user: number,
  unread: number,
}

class RoomUser extends Typegoose implements IRoomUserSchema {
  @prop({ref: 'users'})
  public user: number
  @prop()
  public unread: number
}

class Room extends Typegoose implements IRoomSchema {
  public _id: Types.ObjectId
  @prop({required: true})
  public name: string
  @prop({required: true})
  public type: EnumRoomType
  @arrayProp({items: RoomUser, required: true})
  public users: IRoomUserSchema[]
  @prop({ref: 'users', required: true})
  public ownerId: number
  @prop()
  public avatar: string
}
```
I got an error 
```
/builds/chipwing/chipcoo-im-node/node_modules/mongoose/lib/schema.js:663
    throw new TypeError('Undefined type `' + name + '` at `' + path +
          ^
TypeError: Undefined type `undefined` at `required`
  Did you try nesting Schemas? You can only nest using refs or arrays.
```

After step by step debugging, I found that mongoose got the path `user` schema with `required` options `{required: true, users: ,unread: }`, instead of `{users: ,unread: }`

**Comment by hasezoey** [comment]
In your case you might want to use "Ref\<T>", because Users would be stored in another collection, and just the id's would be stored on the room?

```ts
import * as mongoose from "mongoose";
import { arrayProp, prop, Typegoose, Ref } from "./typegoose";

class RoomUser extends Typegoose {
    @prop({ required: true })
    public user!: number;
    @prop()
    public unread!: number;
}

class Room extends Typegoose {
    @prop({ required: true })
    public name!: string;

    // @prop({ required: true })
    // public type!: EnumRoomType; // disabled because "EnumRoomType" was not provided

    @arrayProp({ itemsRef: RoomUser, required: true })
    public users!: Ref<RoomUser>[];

    @prop({ ref: RoomUser, required: true })
    public ownerId!: Ref<RoomUser>;

    @prop()
    public avatar!: string;
}

const RoomModel = new Room().getModelForClass(Room);
const RoomUserModel = new RoomUser().getModelForClass(RoomUser);

(async () => {
    await mongoose.connect(`mongodb://mongodb:27017/`, {
        useNewUrlParser: true,
        useFindAndModify: true,
        useCreateIndex: true,
        dbName: "verify193",
        user: "user",
        pass: "passwd",
        authSource: "admin",
        autoIndex: true,
    });

    const owner = await new RoomUserModel({
        unread: 0,
        user: 0,
    } as RoomUser);
    await owner.save();

    const instance = await new RoomModel({
        name: "hello",
        ownerId: owner._id,
        avatar: "avatar?",
        users: [owner._id]
    } as Room);
    await instance.save();

    console.log(await RoomModel.findById(instance._id).populate("ownerId users").exec());

    await mongoose.disconnect();
})();

```

PS:
- you didnt provide "EnumRoomType"
- your last line of the issue is wrongly formatted

**Comment by bangbang93** [comment]
users are in another database, not in MongoDB. So I only store user id in the document, and cannot use reference for populate.
RoomUser is a nested document in RoomModel.

**Comment by hasezoey** [comment]
> users are in another database, not in MongoDB

Sorry, didnt know that (it wasnt in the issue)
so i would recommend using virtual properties
```ts
@prop()
public get users() {
    return await otherdb.find();
}
```

**Comment by hasezoey** [comment]
i think this can be closed as "solved" / "stale" / "fixed in v6.0.0?"

@Ben305
